lua/keymappings.lua

```diff
@@ -1,86 +1,108 @@
+    end
+  end
+end
 
--- TODO fix this
--- Terminal window navigation
-vim.cmd [[
-  tnoremap <C-h> <C-\><C-N><C-w>h
-  tnoremap <C-j> <C-\><C-N><C-w>j
-  tnoremap <C-k> <C-\><C-N><C-w>k
-  tnoremap <C-l> <C-\><C-N><C-w>l
-  inoremap <C-h> <C-\><C-N><C-w>h
-  inoremap <C-j> <C-\><C-N><C-w>j
-  inoremap <C-k> <C-\><C-N><C-w>k
-  inoremap <C-l> <C-\><C-N><C-w>l
-  tnoremap <Esc> <C-\><C-n>
-]]
+local mappings = {
+  n = {
+    -- better window movement
+    { "<C-h>", "<C-w>h", { silent = true } },
+    { "<C-j>", "<C-w>j", { silent = true } },
+    { "<C-k>", "<C-w>k", { silent = true } },
+    { "<C-l>", "<C-w>l", { silent = true } },
 
--- TODO fix this
--- resize with arrows
-if vim.fn.has("mac") == 1 then
-  vim.api.nvim_set_keymap("n", "<A-Up>", ":resize -2<CR>", { silent = true })
-  vim.api.nvim_set_keymap("n", "<A-Down>", ":resize +2<CR>", { silent = true })
-  vim.api.nvim_set_keymap("n", "<A-Left>", ":vertical resize -2<CR>", { silent = true })
-  vim.api.nvim_set_keymap("n", "<A-Right>", ":vertical resize +2<CR>", { silent = true })
-else
-  vim.api.nvim_set_keymap("n", "<C-Up>", ":resize -2<CR>", { silent = true })
-  vim.api.nvim_set_keymap("n", "<C-Down>", ":resize +2<CR>", { silent = true })
-  vim.api.nvim_set_keymap("n", "<C-Left>", ":vertical resize -2<CR>", { silent = true })
-  vim.api.nvim_set_keymap("n", "<C-Right>", ":vertical resize +2<CR>", { silent = true })
-end
+    -- resize with arrows
+    { "<C-Up>", ":resize -2<CR>", { silent = true } },
+    { "<C-Down>", ":resize +2<CR>", { silent = true } },
+    { "<C-Left>", ":vertical resize -2<CR>", { silent = true } },
+    { "<C-Right>", ":vertical resize +2<CR>", { silent = true } },
 
--- better indenting
-vim.api.nvim_set_keymap("v", "<", "<gv", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("v", ">", ">gv", { noremap = true, silent = true })
+    -- Tab switch buffer
+    { "<TAB>", ":bnext<CR>" },
+    { "<S-TAB>", ":bprevious<CR>" },
 
--- I hate escape
-vim.api.nvim_set_keymap("i", "jk", "<ESC>", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("i", "kj", "<ESC>", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("i", "jj", "<ESC>", { noremap = true, silent = true })
+    -- Move current line / block with Alt-j/k a la vscode.
+    { "<A-j>", ":m .+1<CR>==" },
+    { "<A-k>", ":m .-2<CR>==" },
 
--- Tab switch buffer
-vim.api.nvim_set_keymap("n", "<TAB>", ":bnext<CR>", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("n", "<S-TAB>", ":bprevious<CR>", { noremap = true, silent = true })
+    -- QuickFix
+    { "]q", ":cnext<CR>" },
+    { "[q", ":cprev<CR>" },
 
--- Move selected line / block of text in visual mode
-vim.api.nvim_set_keymap("x", "K", ":move '<-2<CR>gv-gv", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("x", "J", ":move '>+1<CR>gv-gv", { noremap = true, silent = true })
+    -- {'<C-TAB>', 'compe#complete()', {noremap = true, silent = true, expr = true}},
+  },
+  v = {
```

Comment:
Those variables should be named better than this.

Reply:
Like this v =

Reply:
I don't understand what you mean, what is wrong here ?

Reply:
the name you gave to the variables, like "v" and others. It could be a little more descriptive like "visual" for example.

Reply:
The goal is to follow the same pattern as which-key, also understand that these keys are used as parameters for nvim_set_keymap

Reply:
As a compromise, I'll add a small comment nest to the key

Reply:
ah ok, sounds good